contracts/InterchainTokenService.sol

```diff
@@ -641,23 +640,25 @@ contract InterchainTokenService is
     }
 
     function contractCallWithTokenValue(
-        string calldata /*sourceChain*/,
-        string calldata /*sourceAddress*/,
-        bytes calldata /*payload*/,
-        string calldata /*symbol*/,
-        uint256 /*amount*/
-    ) public view virtual returns (address, uint256) {
-        revert ExecuteWithTokenNotSupported();
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload,
+        string calldata symbol,
+        uint256 amount
+    ) public view virtual onlyRemoteService(sourceChain, sourceAddress) whenNotPaused returns (address, uint256) {
+        _checkPayloadAgainstGatewayData(payload, symbol, amount);
```

Comment:
this method assumes that the payload is for a transfer and returns incomprehensible error if decoding fails on a different type. Since `_callContractValue` returns the address and amount, you could check that directly instead of doing this as well. Alternatively, if we want to reuse the method, then just move it below _contractCallValue which checks the payload type first